app/lib/covid19/candidate_email_delivery.rb

```diff
@@ -0,0 +1,26 @@
+module Covid19
+  class CandidateEmailDelivery
+    def send_delay_emails
+      skip_email_statuses = %w(
+       withdrawn rejected declined conditions_not_met recruited enrolled
+      )
+
+      # Find all candidates that need emailing
+      candidates = Candidate.includes(application_forms: [:application_choices])
+
+      candidates_to_email = candidates.select do |candidate|
+        application_form = candidate.current_application
+        if application_form.application_choices.all? { |ac| ac.status.in? skip_email_statuses }
+          false
+        else
+          true
+        end
+      end
```

Comment:
This is a Ruby array `.select` right? So it'll be slower than running in the database. We've only got 500 candidates on prod so I don't think this is likely to be a bottleneck...

Reply:
Yep, opted for Ruby as it's arguably clearer than the equivalent SQL and total record numbers are low.